### PR TITLE
Fix the flush_size option to work with NG Pipeline

### PR DESCRIPTION
**Pull request by andrewvc**
@suyograo would you be able to have a look at this at some point? Not sure if this is in time to make it with 2.2, if not, no biggie. The 500 flush size default means that unless you set `-b` above 500 this will have no effect anyway.

### CODE REVIEW

File: lib/logstash/outputs/elasticsearch/common.rb [hunk 1]
```diff
@@ -25,7 +25,9 @@ def receive(event)
 
     # Receive an array of events and immediately attempt to index them (no buffering)
     def multi_receive(events)
-      retrying_submit(events.map {|e| event_action_tuple(e) })
+      events.each_slice(@flush_size) do |slice|
```
Comment (suyograo) [inline lib/logstash/outputs/elasticsearch/common.rb]: Shouldn't there be a test if `batch_size > @flush_size` and then slice it? Or does `each_slice` takes care of that internally using the `size` of `events` ?

Comment (andrewvc) [inline lib/logstash/outputs/elasticsearch/common.rb]: `each_slice` does the right thing!

```
irb(main):002:0> [1,2,3,4].each_slice(5).to_a
=> [[1, 2, 3, 4]]
```

Comment (suyograo) [inline lib/logstash/outputs/elasticsearch/common.rb]: nice!

**Comment by suyograo** [comment]
@andrewvc LGTM, but this is late for 2.2 packaging, i've cut the release already. We can go ahead and publish this plugin though

File: lib/logstash/outputs/elasticsearch/common.rb (hunk 1 above)
Comment (suyograo) [inline lib/logstash/outputs/elasticsearch/common.rb]: Can you add a comment here, saying why we are doing this in relationship to `batch_size`?

**Pull request closed without merging**